File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        if(nums.size() < 2) return false;
        vector<pair<int,int>> vec;
        for(int i = 0; i < nums.size(); i++){
            vec.push_back({nums[i], i});
        }
        sort(vec.begin(), vec.end());
        for(int i = 0; i < vec.size() - 1; i++){
            if(vec[i].first == vec[i+1].first && abs(vec[i].second - vec[i+1].second) <= k){
                return true;
            }
        }
        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        //  Dictionary를 이용한 방법
        // Dictionary<int, int> map = new Dictionary<int, int>();
        // for(int i = 0; i < nums.Length; i++){
        //     if(map.ContainsKey(nums[i]) && Math.Abs(map[nums[i]] - i) <= k){
        //         return true;
        //     }
        //     map[nums[i]] = i;
        // }
        // return false;

        //  sliding Window
        for(int i = 0; i < nums.Length; i++){
            int j = i + 1;
            while(j < nums.Length && j - i <= k){
                if(nums[i] == nums[j]){
                    return true;
                }
                j++;
            }
        }
        return false;
    }
}